corehq/apps/api/resources/v0_5.py

```diff
@@ -57,6 +57,18 @@ def user_es_call(domain, q, fields, size, start_at):
     return query.run().hits
 
 
+def _set_role_for_bundle(kwargs, bundle):
+    # check for roles associated with the domain
+    domain_roles = UserRole.by_domain_and_name(kwargs['domain'], bundle.data.get('role'))
+    if domain_roles:
+        qualified_role_id = domain_roles[0].get_qualified_id()
+        bundle.obj.set_role(kwargs['domain'], qualified_role_id)
+    else:
+        # check for preset roles and now create them for the domain
+        permission_preset_name = UserRole.get_preset_permission_by_name(bundle.data.get('role'))
+        if permission_preset_name:
+            bundle.obj.set_role(kwargs['domain'], permission_preset_name)
```

Comment:
it looks like this does nothing if the role is not valid or a preset, and I think that should probably fail. maybe add a test case and then make sure we're handling that